ansible_mitogen/transport_config.py

```diff
@@ -443,7 +443,7 @@ def become_method(self):
         return self._play_context.become_method
 
     def become_user(self):
-        return self._play_context.become_user
+        return self._connection_option('become_user')
```

Comment:
In Ansible `'become_user'` is not an option of the connection plugin. It's an option of the become plugin (e.g. https://docs.ansible.com/ansible/latest/collections/ansible/builtin/sudo_become.html#parameter-become_user). So to the extent that this change is working, it is only doing so by accident - relying on the `PlayContext` fallback in `_connect_option()`.

You would need to use the `get_option()` method of the become plugin, which has its own quirks I'm still working to understand.